play-client/src/main/scala/endpoints/EndpointPlayClient.scala

```diff
@@ -28,21 +28,20 @@ class EndpointPlayClient(wsClient: WSClient)(implicit ec: ExecutionContext) exte
   /**
     * A function that, given an `A` information and a `WSRequest`, eventually returns a `WSResponse`
     */
-  type RequestEntity[A] = (A, WSRequest) => Future[WSResponse]
+  type RequestEntity[A] = (A, WSRequest) => WSRequest
+
+  override val emptyRequestEntity: RequestEntity[Unit] = { case (_, req) => req }
```

Comment:
I would be happy if you make another PR (and remove the remaining `override` keywords at the same time). But if you don’t have time I will do it some day :)

If it’s not a lazy val we can get initialization issues:

~~~ scala
trait SomeEndpoint extends EndpointAlg {
  val foo = endpoint(post(path, emptyRequestEntity), emptyResponse)
}
~~~

Here we are *using* `emptyRequestEntity` while it is still abstract.

Then, when we mix an interpreter:

~~~ scala
object SomeEndpointRouter extends SomeEndpoint with EndpointPlayRouting {
  // …
}
~~~

The initialization semantic in Scala will first run the initialization code of `SomeEndpoint` and then the initialization code of `EndpointPlayRouting`, so we will refer to `emptyRequestEntity` before it will actually be initialized. (I’m just thinking that we could probably prevent this kind of mistakes by making all `Alg` traits *universal traits*).

Also, do you mind renaming `emptyRequestEntity` to `emptyRequest`, for symmetry with `emptyResponse`?